ultimatepython/classes/encapsulation.py

```diff
@@ -6,32 +6,41 @@
 attributes from outside the class. Instead, users must use methods to access and
 modify attributes.
 """
-import random
+import secrets
+
+# Module-level constanct
+_INVALID_AMOUNT_MESSAGE = "Invalid amount."
+_INSUFFICIENT_BALANCE_MESSAGE = "Insufficient balance."
 
 
 class BankAccount:
     def __init__(self, account_holder_name):
         """
-        In python, a class attribute can be made private by prefixing it with two underscores.
+        In Python, a class attribute can be made private by prefixing it with two underscores.
         This makes it inaccessible to users outside the class.
         By default, class attributes are public. Therefore, they can be accessed and modified
         outside of the class.
 
-        Here, account_number and balace are private while account_holder_name is public.
+        Here, account_number and balance are private while account_holder_name is public.
         """
         self.account_holder_name = account_holder_name
         """
-        The account number is generated automaticallyusing the randint function from
+        The account number is generated automatically using the randbelow function from
         the random module when a new instance of the class is created.
-        The balance is set to 0.0 by default.
+        The balance is set to 0 by default.
         """
-        self.__account_number = random.randint(1000000000, 9999999999)  # generate a random account number of 10 digits.
-        self.__balance = 0.0
+        self.__account_number = secrets.randbelow(10**10)  # generate a random account number of 10 digits.
+        self.__balance = 0
 
     def deposit(self, balance):
-        self.__balance += balance  # add the deposited amount to the balance.
+        self.__balance += int(balance)  # add the deposited amount to the balance.
 
     def withdraw(self, balance):
+        # raise a value error if the balance is less than the amount to be withdrawn.
```

Comment:
Can we have a proper docstring for this method?

Reply:
Fixed.